akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -304,38 +313,90 @@ private[akka] class Shard(
 }
 
 /**
+ * INTERNAL API: Common things for PersistentShard and DDataShard
+ */
+private[akka] trait RememberingShard { selfType: Shard ⇒
+  import ShardRegion.{ EntityId, Msg }
+  import Shard._
+  import akka.pattern.pipe
+
+  protected val settings: ClusterShardingSettings
+
+  protected val rememberedEntitiesRecoveryStrategy: EntityRecoveryStrategy = {
```

Comment:
the things in this trait previously existed in `PersistentShard`, pretty much moved without change